sormas-ui/src/main/java/de/symeda/sormas/ui/statistics/StatisticsView.java

```diff
@@ -973,11 +969,50 @@ public void generateMap() {
 					polygon.setCaption(regionOrDistrict.getCaption() + "<br>" + regionOrDistrictValue);
 				}
 				// fillOpacity is used, so we can still hover the region
-				polygon.setOptions("{\"stroke\": false, \"color\": '" + fillColor + "', \"fillOpacity\": " + fillOpacity + "}");
+				polygon.setOptions(
+					"{\"stroke\": true, \"color\": '#000000', \"weight\": 1, \"fillColor\": '" + fillColor + "', \"fillOpacity\": " + fillOpacity
+						+ "}");
 				polygon.setLatLons(shapePart);
 				resultPolygons.add(polygon);
 			}
 		}
+		// sort polygon array, so that polygons which are completely contained by another appear on top
+		List<Integer[]> indexesToSwap = new ArrayList<>();
+		int poly1index = 0, poly2index = 0;
+		for (LeafletPolygon poly1 : resultPolygons) {
+			for (LeafletPolygon poly2 : resultPolygons) {
+				if (poly1index == poly2index) {
+					continue;
+				}
+				// determine maximum latitude and longitude of each polygon
+				List<double[]> poly1LatLons = Arrays.asList(poly1.getLatLons());
```

Comment:
I suggest to give the LeaftletPolygon two fields minLatLon and maxLatLon and a getter method. Whenever the getter is called the first time, calculate the values. Reset them to null, when setLatLons is called.

Thisway the min/max does not have to be calculated for every combination of polygons, but just for every single polygon.